examples/movement/physics_in_fixed_timestep.rs

```diff
@@ -0,0 +1,209 @@
+//! advance a physics simulation in a fixed timestep, and display the results.
+//!
+//! The classic source for how and why this is done is Glenn Fiedler's article
+//! [Fix Your Timestep!](https://gafferongames.com/post/fix_your_timestep/).
+//!
+//! ## Motivation
+//!
+//! The naive way of moving a player is to just update their position like so:
+//! ```no_run
+//! transform.translation += velocity;
+//! ```
+//! The issue here is that the player's movement speed will be tied to the frame rate.
+//! Faster machines will move the player faster, and slower machines will move the player slower.
+//! In fact, you can observe this today when running some old games that did it this way on modern hardware!
+//! The player will move at a breakneck pace.
+//!
+//! The more sophisticated way is to update the player's position based on the time that has passed:
+//! ```no_run
+//! transform.translation += velocity * time.delta_seconds();
+//! ```
+//! This way, velocity represents a speed in units per second, and the player will move at the same speed
+//! regardless of the frame rate.
+//!
+//! However, this can still be problematic if the frame rate is very low or very high.
+//! If the frame rate is very low, the player will move in large jumps. This may lead to
+//! a player moving in such large jumps that they pass through walls or other obstacles.
+//! In general, you cannot expect a physics simulation to behave nicely with *any* delta time.
+//! Ideally, we want to have some stability in what kinds of delta times we feed into our physics simulation.
+//!
+//! The solution is using a fixed timestep. This means that we advance the physics simulation by a fixed amount
+//! at a time. If the real time that passed between two frames is less than the fixed timestep, we simply
+//! don't advance the physics simulation at all.
+//! If it is more, we advance the physics simulation multiple times until we catch up.
+//! You can read more about how Bevy implements this in the documentation for
+//! [`bevy::time::Fixed`](https://docs.rs/bevy/latest/bevy/time/struct.Fixed.html).
+//!
+//! This leaves us with a last problem, however. If our physics simulation may advance zero or multiple times
+//! per frame, there may be frames in which the player's position did not need to be updated at all,
+//! and some where it is updated by a large amount that resulted from running the physics simulation multiple times.
+//! This is physically correct, but visually jarring. Imagine a player moving in a straight line, but depending on the frame rate,
+//! they may sometimes advance by a large amount and sometimes not at all. Visually, we want the player to move smoothly.
+//! This is why we need to separate the player's position in the physics simulation from the player's position in the visual representation.
+//! The visual representation can then be interpolated smoothly based on the last and current actual player position in the physics simulation.
```

Comment:
I think this could mention that this introduces a 1 FixedUpdate visual delay.

Slow FixedUpdate:  FU1 -  F1  - F2  - F3  - FU2 - F4 
At F4, you might have `overstep_fraction = 0.25`, and what you see visually will basically be F1 (an interpolation between FU1 and FU2). Which is what you want, because otherwise the entity would just get updated every `FixedUpdate`, so once every 3-4 frames. With this kind of interpolation, the entity gets updated smoothly every frame.

Fast FixedUpdate (if timestep > 60Hz): FU1 - FU2 - F1 - FU3 - F2
At F2, you will visually see something the value that is interpolated at `overstep_fraction` between FU2 and FU3. Again this is what you want, because otherwise you could easily have the visuals advance by 2 steps in some frames, and by 1 step in another frame

Reply:
Added a disclaimer: [`b405744` (#14223)](https://github.com/bevyengine/bevy/pull/14223/commits/b4057446972176fc9c4401f232445406aba4603a)